Double Quotes Not Escaped Properly

imanickam opened the pull request:
Pull Request for Issue # .
No issue tracker for this.

### Summary of Changes

Double Quotes Not Escaped Properly in the Language String COM_INSTALLER_NO_INSTALLATION_PLUGINS_FOUND

Original Language String:
COM_INSTALLER_NO_INSTALLATION_PLUGINS_FOUND="No installation plugin has been enabled. At least one must be enabled to be able to use the installer. Go to the <a href='index.php?option=com_plugins&view=plugins&filter[folder]=installer' title="Plugin Manager">Plugin Manager</a> to enable the plugins."


Proposed Language String:
COM_INSTALLER_NO_INSTALLATION_PLUGINS_FOUND="No installation plugin has been enabled. At least one must be enabled to be able to use the installer. Go to the <a href='index.php?option=com_plugins&view=plugins&filter[folder]=installer' title=\"Plugin Manager\">Plugin Manager</a> to enable the plugins."


Reason:
The double quotes enclosing the text string "Plugin Manager" is not escaped properly wither with "_QQ_" (current usage) or \" (future usage).  Because of this, the HTML generated where this language string is used is shown as below:

<div class="alert-message">No installation plugin has been enabled. At least one must be enabled to be able to use the installer. Go to the <a href="index.php?option=com_plugins&amp;view=plugins&amp;filter[folder]=installer" title="Plugin" manager="">Plugin Manager</a> to enable the plugins.</div>

Notice the new attribute master next to the title attribute.

With the fully escaped double quotes, the HTML would like the following:

<div class="alert-message">No installation plugin has been enabled. At least one must be enabled to be able to use the installer. Go to the <a href="index.php?option=com_plugins&amp;view=plugins&amp;filter[folder]=installer" title="Plugin Manager">Plugin Manager</a> to enable the plugins.</div>


### Testing Instructions

- Login to the site's back-end
- Go to the screen "Plugins" (Extensions => Plugins)
- Disable all the plugins of the type "Installer"
- Go to the screen "Extensions: Install" (Extensions => Manage -> Install"
- You will see a warning notice about no installation plugins are enabled
- Inspect the HTML created for the Warning message


### Expected result

Review the summary of changes


### Actual result

Review the summary of changes


### Documentation Changes Required

None

imanickam committed:
Double Quotes Not Escaped Properly

Double Quotes Not Escaped Properly in the Language String COM_INSTALLER_NO_INSTALLATION_PLUGINS_FOUND

Original Language String:
COM_INSTALLER_NO_INSTALLATION_PLUGINS_FOUND="No installation plugin has been enabled. At least one must be enabled to be able to use the installer. Go to the <a href='index.php?option=com_plugins&view=plugins&filter[folder]=installer' title="Plugin Manager">Plugin Manager</a> to enable the plugins."


Proposed Language String:
COM_INSTALLER_NO_INSTALLATION_PLUGINS_FOUND="No installation plugin has been enabled. At least one must be enabled to be able to use the installer. Go to the <a href='index.php?option=com_plugins&view=plugins&filter[folder]=installer' title=\"Plugin Manager\">Plugin Manager</a> to enable the plugins."


Reason:
The double quotes enclosing the text string "Plugin Manager" is not escaped properly wither with "_QQ_" (current usage) or \" (future usage.  Because of this, the HTML generated where this language string is used is shown as below:

<div class="alert-message">No installation plugin has been enabled. At least one must be enabled to be able to use the installer. Go to the <a href="index.php?option=com_plugins&amp;view=plugins&amp;filter[folder]=installer" title="Plugin" manager="">Plugin Manager</a> to enable the plugins.</div>

Notice the new attribute master next to the title attribute.

With the fully escaped double quotes, the HTML would like the following:

<div class="alert-message">No installation plugin has been enabled. At least one must be enabled to be able to use the installer. Go to the <a href="index.php?option=com_plugins&amp;view=plugins&amp;filter[folder]=installer" title="Plugin Manager">Plugin Manager</a> to enable the plugins.</div>

How to Test:
- Login to the site's back-end
- Go to the screen "Plugins" (Extensions => Plugins)
- Disable all the plugins of the type "Installer"
- Go to the screen "Extensions: Install" (Extensions => Manage -> Install"
- You will see a warning notice about no installation plugins are enabled
- Inspect the HTML created for the Warning

Gitjk commented:
I think for translators using services like Transifex or Crowdin it is still safer to use either single quotes or the html entity for double quotes `&quot;`.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15714">issues.joomla.org/tracker/joomla-cms/15714</a>.</sub>

franz-wohlkoenig commented:
I have tested this item :white_check_mark: successfully on 313b7fcce53dfe85c39b6cd2ddf6eb1a62841197<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15714">issues.joomla.org/tracker/joomla-cms/15714</a>.</sub>

imanickam commented:
> I think for translators using services like Transifex or Crowdin it is still safer to use either single quotes or the html entity for double quotes &quot;.

Good Point...  There should be some standard guidelines for using the quote (either single or double) in the language strings.

infograf768 commented:
html entity can't be used in href

infograf768 commented:
@imanickam 
As this is not urgent and would be the only lang change for 3.7.1 (if I do not mistake), I think we can delay merging until >3.7.1. I nevertheless suggest to use single quotes instead of escaped double quotes in this case.

imanickam committed:
Changed Escaped Double Quotes with Single Quotes

Per the suggestion of infograf768, changed the Escaped Double Quote (ie. \") with Single Quote (on both sides of the text string Plugin Manager that is being used in the title attribute).

Quy commented:
I have tested this item :white_check_mark: successfully on 654f877f12a901ab4332c741447f74750208512f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15714">issues.joomla.org/tracker/joomla-cms/15714</a>.</sub>

franz-wohlkoenig commented:
I have tested this item :white_check_mark: successfully on 654f877f12a901ab4332c741447f74750208512f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15714">issues.joomla.org/tracker/joomla-cms/15714</a>.</sub>

franz-wohlkoenig commented:
RTC after two successful tests.

joo7 commented:
@infograf768 I just read that there might be issues with single quotes in href too (jjk - VirtueMart):
[http://forum.virtuemart.net/index.php?topic=137335.msg481063#msg481063](url)

infograf768 commented:
I have no issue when using single quotes in href. All depends I guess how the href is obtained. In our lang strings, when using them to link to a specific page, they are perfectly OK.

Gitjk commented:
> In our lang strings, when using them to link to a specific page, they are perfectly OK.
Agreed! I think I had copied a syntax error in one of my test cases. Single quotes should work.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15714">issues.joomla.org/tracker/joomla-cms/15714</a>.</sub>
